### ISSUE TITLE
Cloud config mounts don't seem to support nfs

### ISSUE THREAD

**Issue by ryansch** [rancher/os#1620]
**RancherOS Version: (ros os version)**
v0.8.0

**Where are you running RancherOS? (docker-machine, AWS, GCE, baremetal, etc.)**
AWS

I've integrated AWS EFS into our docker cluster.  I tried adding the following to my cloud config:

```yaml
 mounts:
    - ["${efs_server}:/", "/mnt/efs", "nfs4", "nfsvers=4.1,rsize=1048576,wsize=1048576,hard,timeo=600,retrans=2"]
```

This had no effect.

My current workaround is to run it via a container:

```yaml
    nfs-client:
      image: outstand/nfs-client:latest
      restart: always
      labels:
        - io.rancher.os.remove=false
      volumes:
        - /mnt/efs:/mnt/efs:shared
      privileged: true
      net: host
      environment:
        SERVER: ${efs_server}
        SHARE: /
        MOUNTPOINT: /mnt/efs
        FSTYPE: nfs4
        MOUNT_OPTIONS: nfsvers=4.1,rsize=1048576,wsize=1048576,hard,timeo=600,retrans=2
```

**Comment by SvenDowideit** [comment]
just in case - is this possibly related to #1601 ?

**Comment by ryansch** [comment]
I don't think so.  I'm trying to use the default console with nfs4.1 via cloud config.  It failed silently so I'm not sure where it broke.

**Comment by SvenDowideit** [comment]
@ryansch I haven't used NFS with RancherOS yet - do you know if this works with 0.7.1?

**Comment by ryansch** [comment]
@SvenDowideit We went straight from 0.6.x to 0.8.0 because of https://github.com/rancher/os/issues/1382 so I don't have any experience with 0.7.x.

**Comment by SvenDowideit** [comment]
@ryansch ok - did it work in 0.6.x?

**Comment by ryansch** [comment]
@SvenDowideit No idea!  We only just started trying to do NFS mounts.  I probably should have lead with that instead of my previous comment.

**Comment by SvenDowideit** [comment]
ah, ok, there was an issue with nfs3 - #1601 but I like your workaround :)

**Comment by Angatar** [comment]
I'm using NFS with RancherOS and I had to use my own NFS client to make it working.
This might help:
https://github.com/Angatar/docker-nfs-client

**Comment by SvenDowideit** [comment]
see #1506, merged to v1.1.0 - I think I need to work out how to add tests for this though
